dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2587,6 +2587,9 @@ RtpsUdpDataLink::check_heartbeats()
 void
 RtpsUdpDataLink::send_relay_beacon()
 {
+  static const char beacon_msg[4] = { '\0', '\0', '\0', '\0' };
```

Comment:
This is not a valid RTPS Submessage.  The 4-byte array `{1, 0, 0, 0}` is.  See `struct SubmessageHeader` in RtpsCore.idl.